truffle/src/main/ruby/core/proc.rb

```diff
@@ -33,18 +41,33 @@ class Proc
 
   def curry(curried_arity = nil)
     if lambda? && curried_arity
+
       if arity >= 0 && curried_arity != arity
-        raise ArgumentError, "Wrong number of arguments (%i for %i)" % [
-          curried_arity,
-          arity
-        ]
+        raise ArgumentError, "Wrong number of arguments (#{curried_arity} for #{arity})"
       end
 
-      if arity < 0 && curried_arity < (-arity - 1)
-        raise ArgumentError, "Wrong number of arguments (%i for %i)" % [
-          curried_arity,
-          -arity - 1
-        ]
+      if arity < 0
+        max_args = 0
+        min_args = 0
+        has_rest = false
+
+        parameters.each do |p|
+          type = p.first
+          min_args += 1 if type == :req
+          max_args += 0 if type != :block
```

Comment:
It should be a `+= 1` yes. Seems to be a gap in the spec that it doesn't seem to have cases for calling currying procs & lambdas with somewhere between required and total arguments. I'll fix the typo and add to the sepc to nail that stuff.